MVVM Module/Source/Feed Presentation/FeedViewModel.swift

```
@@ -20,12 +20,17 @@ final class FeedViewModel {
 	
 	var onLoadingStateChange: Observer<Bool>?
 	var onFeedLoad: Observer<[FeedImage]>?
+	var onLoadError: Observer<String>?
```

`onLoadError: Observer<String>?` could be `onErrorStateChange: Observer<String?>?`, so the ViewModel decides when to show/hide errors (instead of the view controller). It's often easier to manage state when all changes are centralized.

For example, in the `loadFeed()` method, the ViewModel can call  `onErrorStateChange(.none)` to hide the error before loading the feed.

---

Thanks for the feedback @caiozullo , I fix the code so the viewModel decides to show/hide the error view. What do you think ?